### ISSUE TITLE
[BUG] [KSM/Kubelet] Bugs in templating of metricRelabelings icw cluster{Id|Name}

### ISSUE THREAD

**Issue by hansbogert** [rancher/rancher#43402]
### Rancher Server Setup
- Rancher version: 2.7
- Installation option (Docker install/Helm Chart):
   - If Helm Chart, Kubernetes Cluster and version (RKE1, RKE2, k3s, EKS, etc):
- Proxy/Cert Details:

### Information about the Cluster
N/A

### User Information
N/A

### Provider Information
N/A

### Describe the bug
When templating rancher-monitor with metricRelabelings for kube-state-metrics and the kubelet, the templating fails, see reproduction

### To Reproduce

```
cd charts/rancher-monitoring/102.0.1+up40.1.2
cat << EOF > minimal.yaml
global:
  cattle:
    clusterId: "foobar"
    clusterName: foorbar

kube-state-metrics:
  enabled: true
  prometheus:
    monitor:
      enabled: true
      interval: 60s
      metricRelabelings:
        - action: keep
          regex: kube_configmap_info
          sourceLabels: [__name__]
kubelet:
  enabled: true
  serviceMonitor:
    metricRelabelings:
      - action: keep
        regex: kubelet_node_config_error
        sourceLabels:
        - __name__
EOF

helm template .  -f minimal.yaml

# output:

```

### Actual Result

```
helm template .  -f minimal.yaml
Error: YAML parse error on rancher-monitoring/templates/exporters/kubelet/servicemonitor.yaml: error converting YAML to JSON: yaml: line 25: did not find expected key

Use --debug flag to render out invalid YAML
```

### Expected Result
Would've expected properly rendered yaml.

### Additional context
SURE-7384

**Comment by MKlimuszka** [comment]
Related PR: https://github.com/rancher/charts/pull/3358

**Comment by MKlimuszka** [comment]
The above PR was merged in to 2.7-head, but someone needs to figure out if we need a forward port or not.

**Comment by alexandreLamarre** [comment]
Requires being forward ported to 2.8/2.9

**Comment by tlatino-suse** [comment]
I can see that the fix for this has only been pushed out to the dev-2.7 chart. Will this be pushed out to the production charts anytime soon? I am also wondering if the change will carry forward to the subsequent chart releases. For example, the fix was made in 102.0.2+up40.1.2, but there is a more recent 103.0.2+up40.1.2 and other more recent versions.

Doublechecking, the indent change was clobbered by https://github.com/rancher/charts/commit/381519ffc24625c57dc0d7fdecd97a306fc03d2a.

**Comment by alexandreLamarre** [comment]
@tlatino-suse Yes it is being forward fit soon

**Comment by pankajsqe** [comment]
default metric relabellings for KSM in rancher-monitoring up45.31.1 issue has been fixed and rancher-monitoring:104.0.0+up45.31.1 chart is getting installed successfully.

rancher v2.9.0-alpha7
rancher-monitoring up45.31.1
cluster : v1.30.1+k3s1



![Image](https://github.com/rancher/rancher/assets/149133120/682a5dee-1ce8-44c4-94e5-97526b224da2)
![Image](https://github.com/rancher/rancher/assets/149133120/074ca039-7089-43dd-a653-bd64c240b7f5)
![Image](https://github.com/rancher/rancher/assets/149133120/a6e183e0-6501-4c7d-b790-97fa5cbb0920)
